### PR TITLE
fix: temporary broken user profile due to new 2fa features

### PR DESCRIPTION
**Pull request by netroms**

### CODE REVIEW

**Review by larshelge — APPROVED** [review]

**Review by jbee — APPROVED** [review]

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/security/TwoFactorController.java [hunk 1]
```diff
@@ -80,6 +82,32 @@
...
+        {
+            throw new WebMessageException( conflict( ErrorCode.E3027.getMessage(), ErrorCode.E3027 ) );
+        }
+
+        String appName = systemSettingManager.getStringSetting( SettingKey.APPLICATION_TITLE );
+
+        Map<String, Object> map = new HashMap<>();
+        map.put( "url", "url" );
+
+        return map;
```
Comment (jbee) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/security/TwoFactorController.java:108]: `Map.of` is great for this type of situation
```suggestion
        return Map.of( "url", url );
```

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/security/TwoFactorController.java [hunk 2]
```diff
@@ -80,6 +82,32 @@
...
+    @ResponseBody
+    public Map<String, Object> getQrCode( @CurrentUser User currentUser )
+            throws WebMessageException
+    {
+        if ( currentUser == null )
+        {
+            throw new WebMessageException( conflict( ErrorCode.E3027.getMessage(), ErrorCode.E3027 ) );
+        }
+
+        String appName = systemSettingManager.getStringSetting( SettingKey.APPLICATION_TITLE );
```
Comment (stian-sandvold) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/security/TwoFactorController.java:103]: Have a look at the sonarcloud code smells; This line is not used and can be removed, and you can also resolve the smell with "deprecated" :)

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/security/TwoFactorController.java (hunk 1 above)
Comment (netroms) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/security/TwoFactorController.java:108]: This method will be removed before release, when UI is migrated.

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/security/TwoFactorController.java (hunk 2 above)
Comment (netroms) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/security/TwoFactorController.java:103]: This method will be removed before release, when UI is migrated.

**Pull request merged**